Assets/Scripts/Enemies/Enemy.cs

```diff
@@ -147,80 +145,79 @@ public class Enemy : MonoBehaviour {
             return false;
         }
     }
-  
+
     public void ApplyStatusEffect(StatusEffect statusEffect) {
         activeStatusEffects.Add(statusEffect);
-        
+
         RefreshStatusEffects();
     }
-    
+
     public void RemoveStatusEffect(StatusEffect statusEffect) {
         activeStatusEffects.Remove(statusEffect);
-        
+
         RefreshStatusEffects();
     }
-    
-    public void ApplyTimedStatusEffect(StatusEffect statusEffect, Tower originTower)
-    {
+
+    public void ApplyTimedStatusEffect(StatusEffect statusEffect, Tower originTower) {
         TimedStatusEffect newTimedStatusEffect = new TimedStatusEffect(statusEffect, originTower);
-        
+
         bool statusEffectMissing = true;
-        
+
         foreach (TimedStatusEffect timedStatusEffect in timedStatusEffects) {
```

Comment:
i'd use LINQ here, something like :
var alreadyAppliedEffect = timedStatusEffects.Find(effect => effect.Equals(newTimedStatusEffect))
and then if found, RefreshTimer(), if not found add it to the list